pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -413,6 +413,7 @@ func realScheduleMonitor(c *Cluster, mon *monConfig) (SchedulingResult, error) {
 	d := c.makeDeployment(mon)
 	d.Name += "-canary"
 	d.Spec.Template.ObjectMeta.Name += "-canary"
+	d.Labels["mon_canary"] = "true"
```

Comment:
Labels are still not applied to the pod spec template per both my and Travis' requests.

Reply:
@BlaineEXE : The labes are applied, now they are in the `getlabels` function in the spec file.

Reply:
I don't see any changes in this PR or in existing master code where `mon_canary=true` or `pvc_name=XXXXX` is set in `getLabels`. These labels need applied to both deployment and pod spec.

Reply:
Maybe i don't understand well your request. What i understood was that it was better to have this labels in in the deployment and in the pod spec, and i put this labels in the "getlabels" function to achieve this. What is not set is a value for this labels because the "mon_canary"="true" or "pvc_name" = "whatever" should only be set in the case of the canary pods. 
Do you want to set a default value in this labels?

Reply:
@BlaineEXE , take into account that the name of the PVC only is known when it is created....is possible to pass this information to the getLabels function, but i think that the code is more clear, readable and easy of understand if we assign directly the values to the labels in the moment that we need it. 
This is the same that it was done previously with:

https://github.com/rook/rook/blob/112a746c625d951678bad80a39e6df3517936cbd/pkg/operator/ceph/cluster/mon/mon.go#L414-L415

Apart of this, If we set this labels for the mon canary pods in the getLabels function, we are changing a function that can be used for any mon, in a function that only can be used for canary pods. Yes, we can add parameters, but as i said previously ... i think that this makes the code more cryptic. (at less for me)